src/NumericalAlgorithms/Spectral/Projection.hpp

```diff
@@ -0,0 +1,26 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Matrix.hpp"  // IWYU pragma: keep
+
+/// MAKE THIS SHOW UP IN DOXYGEN COMMIT TO BE DELETED
+namespace Spectral {
+
+/// The portion of an element covered by a mortar.
+enum class MortarSize { Full, UpperHalf, LowerHalf };
+
+/// The projection matrix from a mortar with `num_points_source` LGL
+/// points to an element with `num_points_dest` LGL points.
+const Matrix& projection_matrix_mortar_to_element(
```

Comment:
Should this be strictly applicable to LGL meshes only, or can it be generalised to Gauss quadrature (perhaps including an interpolation) or an arbitrary basis? Perhaps it could be templated on `Dim` and take `Mesh<Dim> mesh_dest` and `Mesh<Dim-1> mesh_source` instead of the `num_points` arguments.

Reply:
I'd meant to comment on that, thank you for reminding me.  Everything in these functions should work for Gauss points by just replacing the "Basis::lgl" calls with whatever interface we go with for the rest of the spectral stuff (and changing the interface of these functions similarly).  I believe the only assumptions that might break are the checks on the valid sizes of matrices.